renku/core/dataset/dataset.py

```diff
@@ -1174,54 +1195,40 @@ def filter_dataset_files(
 
     records = []
     unused_names = set(names)
+
     for dataset in dataset_gateway.get_all_active_datasets():
+        if (names and dataset.name not in names) or (ignore and dataset.name in ignore):
+            continue
+
+        if tag:
+            tags = dataset_gateway.get_all_tags(dataset)
```

Comment:
might be a bit late to realize this, but maybe we should have a `get_dataset_by_tag` method on the gateway? I think this pattern here sprung up in a couple of places

Reply:
I've added it to `core/dataset/tag.py` since we use gateway to interface with the database.